### PR TITLE
accept target field for volume creation

### PR DESCRIPTION
**Pull request by sharanyad**
### Summary
Earlier, target field was ignored for volume creation and mount. Now if target path is specified, the volume is mounted at that location. 

### Testing

New tests cover the changes: yes

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: ecs-init/volumes/ecs_volume_plugin.go [hunk 1]
```diff
@@ -129,11 +129,13 @@ func (a *AmazonECSVolumePlugin) Create(r *volume.CreateRequest) error {
 	}
 
 	// get driver type from options to get the corresponding volume driver
-	var driverType string
+	var driverType, target string
```
Comment (yumex93) [inline ecs-init/volumes/ecs_volume_plugin.go]: Does this mean customer can specify a target path to specify where they want to mount the volume? Or agent is the one who decides the target?

Comment (sharanyad) [reply ecs-init/volumes/ecs_volume_plugin.go]: for EFS, agent doesnt pass in target and plugin creates the mount path. 
This is just added for mostly testing purposes

Comment (yumex93) [reply ecs-init/volumes/ecs_volume_plugin.go]: What kind of testing?

Comment (sharanyad) [reply ecs-init/volumes/ecs_volume_plugin.go]: if we don't want to mess with the original ECS volumes dir. example: the mount path created by plugin is /var/lib/ecs/volumes/<volume-name> . I can override it using the target field and have /foo as mount

**Pull request merged**